src/hooks/useUsersGuildPins.ts

```diff
@@ -8,7 +8,89 @@ import {
   GUILD_PIN_CONTRACTS,
   GuildPinsSupportedChain,
 } from "utils/guildCheckout/constants"
-import { createPublicClient, http } from "viem"
+import { PublicClient, createPublicClient, http } from "viem"
+
+const getUsersGuildPinIdsOnChain = async (
+  balance: bigint,
+  chain: GuildPinsSupportedChain,
+  address: `0x${string}`,
+  client: PublicClient
+) => {
+  const contracts = Array.from({ length: Number(balance) }, (_, i) => ({
+    abi: GUILD_PIN_CONTRACTS[chain].abi,
+    address: GUILD_PIN_CONTRACTS[chain].address,
+    functionName: "tokenOfOwnerByIndex",
+    args: [address, BigInt(i)],
+  }))
+
+  const results =
+    contracts.length > 0
+      ? await client.multicall({
+          contracts: contracts,
+        })
+      : []
+
+  const errors: Error[] = results
+    .filter((result) => result.status === "failure")
+    .map((result) => result.error)
+
+  const pinIds: bigint[] = results
```

Comment:
This type is redundant, it'll have the proper type since you're already casting result as bigint ;)